### ISSUE TITLE
Column formatting is lost when exporting a table from an embedded dashboard

### ISSUE THREAD

**Issue by zbodi74** [metabase/metabase#33727]
### Describe the bug

Downloading a table contents from an embedded dashboard loses custom column formatting:

Original dashboard:
<img width="946" alt="image" src="https://github.com/metabase/metabase/assets/11080638/2dcb5ab3-fb32-4835-8381-e535c0ccd077">

Downloaded from Metabase:
<img width="732" alt="image" src="https://github.com/metabase/metabase/assets/11080638/dae1df9f-5f31-405e-85f9-db60ff46754b">

Downloaded from the embedded dashboard:
<img width="750" alt="image" src="https://github.com/metabase/metabase/assets/11080638/12bd265e-a109-4227-b531-c7e5790146d9">

### To Reproduce

1. Create a saved question with some decimals and timestamps, apply custom formatting, save it, and add it to a new dashboard.
For the attached examples, I used a SQL question on the Sample Database:

```
with t1 as (
 select *
 FROM (
  VALUES
    (1234.05, 2345.05, 4321.05, 0.00, 1234.00, 1234.05, 0.1920, TIMESTAMP '2023-01-01 12:34:56'),
    (2345.30, 3456.30, 2931.30, 0.00, 0.00, 1234.30, 0.00, TIMESTAMP '2023-01-01 12:34:56'),
    (3456.00, 2300.00, 2250.00, 0.00, 122.00, 4250.00, 0.1158, TIMESTAMP '2023-01-01 12:34:56')
  )
), 
t2 as (
select 
    c1 as revenue,
    c2 as ar_profit,
    c3 as ap_profit,
    c4 as dc_profit,
    c5 as profit,
    c6 as cost,
    c7 as margin_ratio,
    c8 as created_at
    from t1
)
select * from t2
```
Then selected table for visualization and customized columns: set decimals to 2 places, added percentage format for `margin_ratio`, set `dc_profit` as hidden, and removed the time part for `created_at`.

2. Enable sharing, and embed the dashboard into a webapp.

3. First download the data as xlsx from the dashboard, directly from the Metabase website. See that the formatting is almost correct. (The time part of the `created_at` column is still visible, but that issue is already covered in https://github.com/metabase/metabase/issues/14393).

4. Then go to dashboard in the embedding webapp, and download the table. Note that the hidden column has reappeared and the formatting is lost.

### Expected behavior

Custom formatting should be preserved and be the same regardless whether downloaded directly from Metabase or via a dashboard embedded into another site.

### Logs

n/a

### Information about your Metabase installation

```JSON
Tested with v1.47.0 .
```

### Severity

It has been reported by a customer and we're not aware of a workaround.

**Comment by WiNloSt** [comment]
There seems to be an option to do this in our API, but it doesn't look like we use this option to process the result.

https://github.com/metabase/metabase/blob/4657ee5a6f08baf8a7f3459c18a8f3396695c818/src/metabase/api/embed.clj#L514

Maybe there is something we missed from https://github.com/metabase/metabase/pull/17226

**Comment by alxnddr** [comment]
Related to https://github.com/metabase/metabase/issues/17753

**Comment by adam-james-v** [comment]
Re-opened because I need to confirm whether or not #36067 fully fixes this.

**Comment by adam-james-v** [comment]
> Note that it also effects column ordering which get ignored when downloading from an embedded setting

Yes, that part _should_ be solved by #36067 but will make sure it's fixed when I add tests/finish solving this one :)
